### PR TITLE
Don't add ":main" to a DID

### PR DESCRIPTION
**Pull request by Thoralf-M**
#72

### CODE REVIEW

**Review by l1h3r — CHANGES_REQUESTED** [review]
I think it would be easier to normalize the DIDs whenever appropriate (we already have a function for DIDs and could add one to normalize documents as well).

If going down the route of preventing people from explicitly specifying `main`, this needs to:

- Remove `normalize/normalized` from `IotaDID` since these will add it back
- Make sure `main` is removed from DIDs created through `try_from_did/parse`
- Note: You can ignore `DerefMut` for now, it will be removed soon (allows direct modification of the `id_segments` field)

**Comment by Thoralf-M** [comment]
https://github.com/iotaledger/identity.rs/blob/87c9ab6c26ff77717bcfbae74973e56c897bc522/identity_iota/src/did/did.rs#L134
So remove "main:" in the `normalize` method and call `normalized` in `try_from_did`?

**Comment by l1h3r** [comment]
> So remove "main:" in the normalize method and call normalized in try_from_did?

Pretty much, it would be better to use `normalize` though and avoid the clone.

Might actually be a good idea to remove `normalized` (returns new clone) and keep `normalize` (mutates self). The two methods can get confusing and it's trivial to call `.clone()` when you need the behaviour of `normalized`.

File: identity_iota/src/did/did.rs [hunk 1]
```diff
@@ -131,8 +131,19 @@ impl IotaDID {
 
     pub fn normalize(&mut self) {
         match &*self.id_segments {
-            [_] => self.id_segments.insert(0, Self::NETWORK.into()),
-            [_, _] | [_, _, _] => {}
+            [_] => {}
+            [network, _] => {
```
Comment (l1h3r) [inline identity_iota/src/did/did.rs:135]: you can use the same style of pattern as before -> `[network, _] | [network, _, _] => {`

Comment (Thoralf-M) [reply identity_iota/src/did/did.rs:135]: Thanks, forgot the shard DID the first time that's why I had them separated

**Review by l1h3r — APPROVED** [review]

File: identity_iota/src/did/did.rs [hunk 2]
```diff
@@ -133,18 +137,16 @@ impl IotaDID {
 
     pub fn normalize(&mut self) {
         match &*self.id_segments {
-            [_] => self.id_segments.insert(0, Self::NETWORK.into()),
-            [_, _] | [_, _, _] => {}
+            [_] => (),
+            [network, _] | [network, _, _] => {
+                if let "main" = network.as_str() {
```
[the remark this answers is not in the document]
Comment (Thoralf-M) [reply identity_iota/src/did/did.rs:142]: Yes

**Pull request merged**